### ISSUE TITLE
sPRUNED: c-lightning does not get funding updates

### ISSUE THREAD

**Issue by Stadicus** [gdassori/spruned#63]
This is probably not a pure sPRUNED issue, but maybe you have some experience with that issue already. I run a sPRUNED bitcoin node with c-lightning 0.6 and from the looks of it everything looks nice. Configuration as follows:
https://gist.github.com/Stadicus/a05c3c5ac6a63cdcfe1aae2b77f17cba

The issue, however, is that c-lightning does not get funding updates. The first send was received by c-lightning, but then no more. On-chain transactions to the `newaddr` are not shown, and incoming channels are visible, but with status `CHANNELD_AWAITING_LOCKIN:They've confirmed funding, we haven't yet.` Meaning, that my c-lightning did not get the onchain funding transaction info.

Is there something special to consider with that constellation?

**Comment by gdassori** [comment]
Hi,

- is the spruned "bitcoin-cli getblockchaininfo" output aligned to the network ?
- is the "getblock <blockhash>" output, with the blockhash related to the given channel, properly working?
- can you recognize something in spruned.log or clightning.log (both in debug mode) that can highlight some ongoing issue ?

Thanks.

**Comment by Stadicus** [comment]
A bit more background: everything runs smoothly on a Raspberry Pi Zero. The issue might that I use an old Bitcoin Core version (0.13.2), I think that c-lightning needs 0.15 at least. Unfortunately, this is the most recent version for the RPiZero architecture (ARMv6) using apt, and I did not have the time yet to cross-compile.

I cannot test thoroughly atm, on my way to the Lightning Hackday in Berlin. If the old bitcoin-cli version is not clearly the issue I'll test next week.

**Comment by Stadicus** [comment]
I've compiled the latest `bitcoin-cli` for the Raspberry Pi Zero (ARMv6) and am currently syncing sPRUNED from scratch. Let's see how that effects the balances shown by c-lightning.

**Comment by gdassori** [comment]
I see, my advice is to wait until spruned is fully synced before starting clightning. Also, if is cool to you that I see your logs, I'm available :-)

**Comment by Stadicus** [comment]
Thanks. Yes, spruned is already up and running, but doing a `lightning-cli dev-rescan-outputs` takes ages (could take more than a week). I'll gladly share my logs once finished, but they're already huge... ;-)

**Comment by Stadicus** [comment]
Similar behaviour on my **Digital Ocean VPS** with sPRUNED, bitcoin-cli v0.16.0 and c-lightning. Stale funding transactions (multpile weeks old), services have been running all the time. Funds cannot be withdrawn, but here the `lightning-cli dev-rescan-outputs` frees the on-chain funds. Not sure about the closing channel yet...

```
bitcoin@trueno-04:~$ lightning-cli listfunds
{
  "outputs": [
    {
      "txid": "62...9",
      "output": 0,
      "value": 4224,
      "address": "bc1...t",
      "status": "unconfirmed"
    },
    {
      "txid": "6f...a",
      "output": 0,
      "value": 98606,
      "address": "bc1qal96q9dr3wu8q7qqhhqjp2qvdjqeca09guxkcm",
      "status": "unconfirmed"
    }
  ],
  "channels": [
    {
      "peer_id": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
      "channel_sat": 130000,
      "channel_total_sat": 130000,
      "funding_txid": "62...9"
    }
  ]
}
bitcoin@trueno-04:~$ lightning-cli listpeers
{
  "peers": [
    {
      "id": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
      "connected": true,
      "netaddr": [
        "167.99.50.31:9735"
      ],
      "alias": "rompert.com🔵",
      "color": "0000ff",
      "channels": [
        {
          "state": "CHANNELD_SHUTTING_DOWN",
          "owner": "lightning_channeld",
          "channel_id": "3955524ceb86a3bad8963714454181bbe66e13ae06396311e935ce626f1f9363",
          "funding_txid": "62...9",
          "msatoshi_to_us": 130000000,
          "msatoshi_to_us_min": 130000000,
          "msatoshi_to_us_max": 130000000,
          "msatoshi_total": 130000000,
          "dust_limit_satoshis": 546,
          "max_htlc_value_in_flight_msat": 18446744073709551615,
          "their_channel_reserve_satoshis": 0,
          "our_channel_reserve_satoshis": 1300,
          "channel_reserve_satoshis": 0,
          "spendable_msatoshi": 128700000,
          "htlc_minimum_msat": 0,
          "their_to_self_delay": 144,
          "our_to_self_delay": 144,
          "to_self_delay": 144,
          "max_accepted_htlcs": 483,
          "status": [
            "CHANNELD_SHUTTING_DOWN:Sent reestablish, waiting for theirs"
          ],
          "in_payments_offered": 0,
          "in_msatoshi_offered": 0,
          "in_payments_fulfilled": 0,
          "in_msatoshi_fulfilled": 0,
          "out_payments_offered": 0,
          "out_msatoshi_offered": 0,
          "out_payments_fulfilled": 0,
          "out_msatoshi_fulfilled": 0
        }
      ]
    }
  ]
}
```

Funds cannot be withdrawn, c-lightning seems out of sync with blockchain:
```
bitcoin@trueno-04:~$ lightning-cli withdraw 3M...1 all
{
  "tx": "02000...00",
  "txid": "{   \"message\": \"the transaction was rejected by network rules.\n\nMissing inputs\n[02000...00]\",   \"code\": 1 }"
}
```

A rescan of outputs results in multiple changes to c-lightning output states:
```
bitcoin@trueno-04:~$ lightning-cli dev-rescan-outputs
{
  "outputs": [
    {
      "txid": "43...8",
      "output": 1,
      "oldstate": 2,
      "newstate": 2
    },
    {
      "txid": "6f...a",
      "output": 0,
      "oldstate": 2,
      "newstate": 0
    },
    {
      "txid": "62...9",
      "output": 0,
      "oldstate": 2,
      "newstate": 2
    },
    {
      "txid": "a9...b",
      "output": 1,
      "oldstate": 2,
      "newstate": 0
    }
  ]
}
bitcoin@trueno-04:~$ lightning-cli listfunds
{
  "outputs": [
    {
      "txid": "a9...b",
      "output": 1,
      "value": 134991,
      "address": "bc1qeve80s8sxh4eqkqw6n0nen0fsm4rxjzcc78lqc",
      "status": "unconfirmed"
    },
    {
      "txid": "6f...a",
      "output": 0,
      "value": 98606,
      "address": "bc1qal96q9dr3wu8q7qqhhqjp2qvdjqeca09guxkcm",
      "status": "unconfirmed"
    }
  ],
  "channels": [
    {
      "peer_id": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
      "channel_sat": 130000,
      "channel_total_sat": 130000,
      "funding_txid": "62...9"
    }
  ]
}
```

Funds can now be withdrawn.
```
bitcoin@trueno-04:~$ lightning-cli withdraw 3M...1 all
{
  "tx": "0200...0",
  "txid": "f4...c"
}
bitcoin@trueno-04:~$ lightning-cli listfunds
{
  "outputs": [
  ],
  "channels": [
    {
      "peer_id": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
      "channel_sat": 130000,
      "channel_total_sat": 130000,
      "funding_txid": "62...9"
    }
  ]
}
bitcoin@trueno-04:~$
```

**Comment by gdassori** [comment]
As far as I know, the dev-rescan-outputs command uses the gettxout api, which is electrum based in spruned, while the normal usage is backed by the getblock api, which is p2p network based, this may (but it's hard, just a guessing) drive into inconsistencies.

Also, I don't use services-based daemons (clightning & spruned with systemd scripts), so when it crashes, here it stops, and I see it. 
If you, instead, are using something to have them restarted when they stops, this may explain why you go into this. 
The usage is very intensive, and maybe at some point clightning is requesting data from spruned which is still syncing (cause both of them restarted) and.. mmhm, weid stuff happens.


My advices are to:

- increase the spruned blocks cache as much as you can (1-2gb should be nice to increase rescans speed. I would say "as much as you can", however, is the best option)

- download the latest spruned version released tonight on pypi, which wraps a more consistent bitcoind API emulation (with error code -1 on generic failures) and should reduce third party apps errors (clightning, in this case) and so rescans.

**Comment by Stadicus** [comment]
Back on the Raspberry Pi Zero. Thanks for your input. I cancelled the old rescan, updated sPRUNED and ran all services manually. The `dev-rescan-outputs` was a matter of seconds. 

Strangely, c-lightning knows about outputs that all went into the one channel this node had. However, this channel is long closed, c-lightning knows that, but the funds are still locked.

What do you think, more of a c-lightning issue or is there anything I can do in regards to sPRUNED?

```
bitcoin@StadicusZero:~ $ lightning-cli dev-rescan-outputs
{
  "outputs": [
    {
      "txid": "ab52ad0814e622521b7519534493ca1c38add630f6da273356df441baac2e64d",
      "output": 0,
      "oldstate": 2,
      "newstate": 2
    },
    {
      "txid": "8a3aec98b23008c19d88497fffecb79ffdcf8431e64000f3025b6770424fddbc",
      "output": 2,
      "oldstate": 2,
      "newstate": 2
    },
    {
      "txid": "309544e23ae5795fd8fc82410279e0e70ee3c86c9734bffed75c21b37ee3f164",
      "output": 2,
      "oldstate": 2,
      "newstate": 2
    }
  ]
}
bitcoin@StadicusZero:~ $ lightning-cli listfunds
{
  "outputs": [
  ],
  "channels": [
    {
      "peer_id": "03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56",
      "short_channel_id": "538243:1179:0",
      "channel_sat": 358313,
      "channel_total_sat": 358313,
      "funding_txid": "58ba38c86b0f5ca4f23f81414851c2abb91d90100f5d7a2c23b65c86bc273606"
    }
  ]
}
bitcoin@StadicusZero:~ $ lightning-cli listpeers
{
  "peers": [
    {
      "id": "03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56",
      "connected": false,
      "channels": [
        {
          "state": "CLOSINGD_COMPLETE",
          "short_channel_id": "538243:1179:0",
          "channel_id": "063627bc865cb6232c7a5d0f10901db9abc2514841813ff2a45c0f6bc838ba58",
          "funding_txid": "58ba38c86b0f5ca4f23f81414851c2abb91d90100f5d7a2c23b65c86bc273606",
          ...
          "status": [
            "Loaded from database"
          ],
          ...
        }
      ]
    }
  ]
}
```

**Comment by coinsmith** [comment]
Sent a small amount to c-lightning running on spruned. After 6 confirmations didn't see the transaction through ./lightning-cli listfunds. Restarted lightning deamon with a --rescan 50 option, funds now show.

**Comment by Stadicus** [comment]
Same here, thanks for the hint. Started `lightningd --rescan 2200` (but aborted after a few hundred) and restarted `lightningd`. The funds were now present, see log output. After that I could withdraw all funds. 

```
2018-09-09T19:39:49.402Z lightningd(17448): Loaded 0 HTLC signatures from DB
2018-09-09T19:39:49.411Z lightningd(17448): Loaded 1 channels from DB
2018-09-09T19:39:49.460Z lightningd(17448): Loading HTLCs for channel 1
2018-09-09T19:39:49.536Z lightningd(17448): Restored 0 incoming and 0 outgoing HTLCS
2018-09-09T19:39:58.615Z lightning_gossipd(17461): Disabling channel 538243:1179:0/0, active 0 -> 0
2018-09-09T19:39:59.071Z lightningd(17448): Adding block 538673: 000000000000000000123f1959913d011e5e5dc8fa61ece30e205a3bf6145fae
2018-09-09T19:39:59.133Z lightningd(17448): Listening on 'lightning-rpc'
2018-09-09T19:39:59.191Z lightning_gossipd(17461): Failed to create 10 socket: Address family not supported by protocol
2018-09-09T19:39:59.193Z lightning_gossipd(17461): Created IPv4 listener on port 9735
2018-09-09T19:39:59.195Z lightning_gossipd(17461): Received node_announcement for node 026bdccbf130442edfe1ea43b3e402a81f7c1b0d6c515b58f79c4a2896ca58ab42
2018-09-09T19:39:59.200Z lightning_gossipd(17461): REPLY WIRE_GOSSIPCTL_ACTIVATE_REPLY with 0 fds
2018-09-09T19:39:59.204Z lightningd(17448): Restarting onchaind for channel 1
2018-09-09T19:39:59.211Z lightningd(17448): 03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: Peer permanent failure in ONCHAIN: Funding transaction spent
2018-09-09T19:39:59.211Z lightningd(17448):      (tx 8c2e06cf5e9f1ffd7d48c10d2e8c33f88918679551a7ee7119930508adcdff28)
2018-09-09T19:39:59.230Z lightningd(17448): sendrawtransaction: 02000000000101063627bc865cb6232c7a5d0f10901db9abc2514841813ff2a45c0f6bc838ba580000000000ffffffff01e27605000000000016001421109906584da8f1b3cce117a575cd7d13ba2b6a0400473044022061406a2dc898a712766837ce4cb780baa6d9497122bb301c7870d83de9492fdd02201a465f3a279f4e0caaa39182126d43c62ec191de320f2d38201c44cbd8147ad601483045022100b78dbb103391ad2d71086873132855b15b75bee249ceb8124b449b4ef3e1ca6f022035e999ad19baa39651a5a773c34a54eafe3d10028a39eb2140719666407893040147522102c536247847e318bc2e871382226ebe2133f92843669ea90e3fcb28d2971980af2102f659bc4550a300cde1c1867417f227ea2b1704424cdffa9f67a56f576dbe4b1d52ae00000000
2018-09-09T19:39:59.237Z lightningd(17448): 03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: State changed from ONCHAIN to FUNDING_SPEND_SEEN
2018-09-09T19:39:59.272Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: pid 17477, msgfd 18
2018-09-09T19:39:59.383Z lightningd(17448): No fee estimate for Normal: basing on default fee rate
2018-09-09T19:39:59.559Z lightningd(17448): Server started with public key 026bdccbf130442edfe1ea43b3e402a81f7c1b0d6c515b58f79c4a2896ca58ab42, alias StadicusZero (color #026bdc) and lightningd v0.6
2018-09-09T19:39:59.740Z lightningd(17448): sendrawtx exit 0, gave {?  "code": 1,?  "message": "the transaction was rejected by network rules.\n\nMissing inputs\n[02000000000101063627bc865cb6232c7a5d0f10901db9abc2514841813ff2a45c0f6bc838ba580000000000ffffffff01e27605000000000016001421109906584da8f1b3cce117a575cd7d13ba2b6a0400473044022061406a2dc898a712766837ce4cb780baa6d9497122bb301c7870d83de9492fdd02201a465f3a279f4e0caaa39182126d43c62ec191de320f2d38201c44cbd8147ad601483045022100b78dbb103391ad2d71086873132855b15b75bee249ceb8124b449b4ef3e1ca6f022035e999ad19baa39651a5a773c34a54eafe3d10028a39eb2140719666407893040147522102c536247847e318bc2e871382226ebe2133f92843669ea90e3fcb28d2971980af2102f659bc4550a300cde1c1867417f227ea2b1704424cdffa9f67a56f576dbe4b1d52ae00000000]"?}?
2018-09-09T19:40:00.727Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: Tracking output 0 of 58ba38c86b0f5ca4f23f81414851c2abb91d90100f5d7a2c23b65c86bc273606: FUNDING_TRANSACTION/FUNDING_OUTPUT
2018-09-09T19:40:00.729Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: Remote per-commit point: 0236a4b7770e922b23d0876c4c85219b4717a532314f0d52396a7b209f6d5a9dd4
2018-09-09T19:40:00.733Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: Old remote per-commit point: 03c3acb72db659b3b1cb44bb22b791acdcaccfde4f76b2c1d5bc3c341ee480c6b0
2018-09-09T19:40:00.745Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: UPDATE WIRE_ONCHAIN_INIT_REPLY
2018-09-09T19:40:00.746Z lightningd(17448): 03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: State changed from FUNDING_SPEND_SEEN to ONCHAIN
2018-09-09T19:40:00.820Z lightningd(17448): lightning_onchaind-03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56 chan #1: Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by MUTUAL_CLOSE (10b784ab2c1e2bb8baca3a96c4356f9cfc01c59510851ab8c34dd25f03764713)
```

At the moment I would not really trust this sPRUNED / c-lightning setup, to be honest. Can I provide information to make this better, eg. logfiles or try specific `bitcoin-cli` commands?

**Comment by gdassori** [comment]
Yes, looks like something there in spruned, there, was hanging. 
Please note the codebase is immature and the userbase is small, neither me would bet on spruned reliability.

Not sure if related but I would say this:
```
2018-09-09T19:39:59.230Z lightningd(17448): sendrawtransaction: [.....]
```
is in line with the described situation: the wallet utxo is not up to date, and must be cause of a missing notification, just before the `State changed from ONCHAIN to FUNDING_SPEND_SEEN`
The outpoint of the rawtransaction didn't existed anymore.

I'm trying to reproduce the bug, it must be a race condition.
Your spruned logs would be very helpful at this point. You can gzip and mail them to guido[dot]dassori at gmail if you wish.

`--rescan 2200` on spruned should be some sort of "atomic option", it's very intensive and you may be going to download >2gb of data from the P2P network, if you hadn't a large cache, it's reasonable that it takes a lot.

**Comment by coinsmith** [comment]
Channel funding error, balance is over 0.001.

./lightning-cli fundchannel 0304b3a9d0f8446a50944d9e317185460361d3bdbd563214faa751585ba46428a5 35000
{ "code" : 301, "message" : "Cannot afford transaction" }

**Comment by coinsmith** [comment]
Managed to open 2 channels, waiting for confirmations.

**Comment by gdassori** [comment]
Stadicus gave me a lot of useful info, thanks for the logs.

Looks like at some point the peers he's connected to refuses to answer to a specific InvItem request, the one for the block 000000000000000000204c6219796604394a43bb765bdd25a39e9eb7aa7d2cbb fails on any connected peer. [I really have to investigate why *this* block, not the one before, not the one after, is hard to be fetched]

The previous peers implementation was naive: spruned didn't add new peers in the pure p2p way, but instead used only the ones available on the dns seed.

The new release, 0.0.2a7, implement the getaddr method on the p2p layer, and so now spruned collect, persist and uses new peers once discovered.

I've also reduced timeouts, re-implemented the peers ban when too many failures happened which, at some point, disappeared :-(.

Would you please try it and see if things goes better.

**Comment by Stadicus** [comment]
Thanks @gdassori for your effort. I checked out the latest release 0.0.2a7 from github and reinstalled the whole Python virtual env from scratch. As I do not have Python 3.6 on the RasPi Zero, I used Python 3.5.3. Installation went through without issues.

When starting sPRUNED, I get the following error message:

```
(venv) bitcoin@StadicusZero:~/src/spruned $ venv/bin/spruned --network bitcoin.mainnet --rpcuser xxx --rpcpassword xxx --cache-size 8000
Traceback (most recent call last):
  File "venv/bin/spruned", line 11, in <module>
    load_entry_point('spruned==0.0.2a6', 'console_scripts', 'spruned')()
  File "/home/bitcoin/src/spruned/venv/lib/python3.5/site-packages/spruned-0.0.2a6-py3.5.egg/spruned/app.py", line 87, in main
    from spruned.main import main_task
  File "/home/bitcoin/src/spruned/venv/lib/python3.5/site-packages/spruned-0.0.2a6-py3.5.egg/spruned/main.py", line 4, in <module>
    from spruned.builder import cache, headers_reactor, blocks_reactor, jsonrpc_server, repository
  File "/home/bitcoin/src/spruned/venv/lib/python3.5/site-packages/spruned-0.0.2a6-py3.5.egg/spruned/builder.py", line 34, in <module>
    jsonrpc_server, headers_reactor, blocks_reactor, repository, cache = builder(_ctx)
  File "/home/bitcoin/src/spruned/venv/lib/python3.5/site-packages/spruned-0.0.2a6-py3.5.egg/spruned/builder.py", line 14, in builder
    electrod_connectionpool, electrod_interface = electrod_builder(ctx)
  File "/home/bitcoin/src/spruned/venv/lib/python3.5/site-packages/spruned-0.0.2a6-py3.5.egg/spruned/daemon/electrod/__init__.py", line 40, in build
    connections=ctx.max_electrum_connections,
AttributeError: 'Context' object has no attribute 'max_electrum_connections'
```

**Comment by gdassori** [comment]
Please check out the 0.0.2a8

**Comment by Stadicus** [comment]
Nice, starting up and beginning bootstrap 200 blocks verification. Will only take about an hour... :-)

**Comment by gdassori** [comment]
Glad to hear :-) 
To be honest I'm beginning to be skeptical about the bootstrap feature, especially on low end devices. 
Probably it makes sense to set it at a very low value by default.

**Comment by Stadicus** [comment]
Is it possible, that the bootstrap process efficiency has decreased with the update? I had in mind that it took an hour, or maybe two, but now after running for 8 hours it's at block 76.

**Comment by gdassori** [comment]
Yes, I noticed that the effort to manage the peers was excessive for the RPi0, and reduced the dataset. There's still room for improvements on the peers management but please checkout the latest build, the issue is fixed, I did ~400blocks\hour on my rpi0 tonight.

**Comment by Stadicus** [comment]
Performance seems still pretty slow. I updated today morning, the bootstrap started at block 53 and now (12h later) it's at block 92. RasPi Zero runds at full capacity (load avg >2), but blocks are being validatet very irregularly. I also noted two tracebacks.

Here's the log: https://pastebin.com/raw/KdQSngF1

I can open a new issue for that, if you'd like, as we strayed quite a bit from the original topic.

**Comment by gdassori** [comment]
Oh gosh, I'm pretty sure your ~/.spruned/p2p_peers.json went too big due the past issue, and you're still in the same situation, despite no new peers are flooded in.

You can stop spruned, delete the file and then restart, and see. 
I'm handling the case, however.

And yes, any new issue is welcome, please :-)

**Comment by Stadicus** [comment]
Doing that right now. Next issue I'll open a new one, promise. :-)

![grafik](https://user-images.githubusercontent.com/19550140/45445361-4d388980-b6ca-11e8-99ea-5570d7ed7358.png)

Looks much better now:
```
2018-09-12 19:47:39,779 - p2p - INFO - Bootstrap: saved block 00000000000000000005aa2809af21eb96bea55d50fd5064da9fe308045c5e4e (102/200)
2018-09-12 19:47:57,695 - electrum - INFO - Last processed header: 541133 (000000000000000000260ac99896784f08f45a31439a2ff1d6536a60044491f5)
2018-09-12 19:48:50,164 - p2p - INFO - Connected to peer 188.40.116.75:8333 (Satoshi:0.15.1)
2018-09-12 19:48:50,516 - p2p - INFO - Connected to peer 66.96.199.249:8333 (Satoshi:0.16.2)
2018-09-12 19:49:30,316 - p2p - INFO - Bootstrap: saved block 00000000000000000001e651ede56f48acccb70ab673407604eece4dc26c2ae3 (103/200)
2018-09-12 19:50:33,490 - cache - INFO - Cache is ok, size: 0, limit: 8000
2018-09-12 19:50:51,674 - p2p - INFO - Connected to peer 172.104.214.233:8333 (Satoshi:0.13.2)
2018-09-12 19:51:23,106 - p2p - INFO - Bootstrap: saved block 00000000000000000010593bce0af6f3ae181b7f178a2ccba219cf191d6d29e6 (104/200)
2018-09-12 19:52:13,854 - p2p - INFO - Bootstrap: saved block 00000000000000000018ee70263b9d9d03a5c1bb3d1ce2c4c893b9fed24334c6 (105/200)
```

**Comment by gdassori** [comment]
Oh, great. Once you establish a reliable connections pool it should go even better. Hopefully... :p

**Comment by Stadicus** [comment]
Back to the original topic, the Pi Zero runs seemingly smoothly, but I noticed that c-lightning is stuck at block 

```
bitcoin@StadicusZero:~ $ lcli getinfo
{
  "id": "031e8f8526e135eae7b1e8fdb60f57222d0353366e4981ad1f43a84750377dbdda",
  "port": 9735,
  "address": [
  ],
  "binding": [
    {
      "type": "ipv4",
      "address": "0.0.0.0",
      "port": 9735
    }
  ],
  "version": "v0.6",
  "blockheight": 541222,
  "network": "bitcoin"
}
bitcoin@StadicusZero:~ $ bitcoin-cli getblockchaininfo
{
  "warning": "spruned 0.0.2a9, emulating bitcoind v0.16",
  "chainwork": null,
  "verificationprogress": 100,
  "bestblockhash": "0000000000000000001a862e4d74646fb32a3f688e4c39c38fd97500a38ada26",
  "blocks": 541512,
  "headers": 541512,
  "pruned": false,
  "mediantime": 1537011950,
  "difficulty": null,
  "chain": "main"
}
```

I attached my log files in case you want to check them. 
[spruned.log.tar.gz](https://github.com/gdassori/spruned/files/2385663/spruned.log.tar.gz)

**Comment by gdassori** [comment]
Howdy,
While spruned was working properly, looks like clightning stopped to poll for new blocks, also if there is no evidence about that in the logs: while I see a successful answered 'getblockhash 541223' call in the spruned logs, there's no 'getblock 0000000000000000001539837894ee508dfb607cd117000ed4dfa5fba12b6a38' call.

A spruned version (0.0.2b1) newer than the one I see in the logs, should reduce network errors and improve performances, as the new connectionspool is smarter, but it still may happen that a call fails (a simple network outage), so I'm afraid those kind of errors have to be addressed on the other side.

**Comment by gdassori** [comment]
Hi @Stadicus , is this an issue ?

**Comment by Stadicus** [comment]
Hej @gdassori, no big deal for me. My takeaway is that the combination as a whole is not stable enough yet, sadly. But nothing sPRUNED could do at this time, I'll just try again with c-lightning 0.7 or 0.8... :-)
